### ISSUE TITLE
Are we too flexible?

### ISSUE THREAD

**Issue by peelyKoodoo** [KoodooHQ/K2#27]
https://github.com/KoodooHQ/K2/pull/19#issuecomment-989893199

My hot take thoughts are no we are not. We want our code to be able to do whatever the dev might need, and whatever design might want in the future. It's easier to be flexible now, and strict in design review, than it is be strict now, and open up the flexibility later if it's asked for.

**Comment by stevelouiscooper** [comment]
I don't think so. I think between using props and composed components we're locking down components enough that any conscientious developer can see the implementation details in `bit.dev` and avoid making any mistakes with the component implementation. I've noticed that we don't have all of the prop descriptions populated which would be good as it would indicate we've thought about why we've added a prop and what it's intended to be used for to the developer researching the component, but other than that I think we kind of have to wait and see how engineers actually end up using the components to see if we've been too flexible with the implementation. My thoughts on this are essentially: If we see component permutations that are not in Figma we've been too flexible and we should be leveraging Typescript and the documentation pieces bit.dev displays to encounter and resolve these issues early in the build phase.

**Comment by peelyKoodoo** [comment]
> I've noticed that we don't have all of the prop descriptions populated which would be good

Yup, we have to document everything. No random props. I saw `showPercentage` earlier and I had to read the code to be sure what it meant. To me it could repeats either the entire bar, the percentage number, or the percentage symbol. It was option 2, so perhaps "showPercentageLabel` would be better, but yes a full description would be even better still.

> wait and see how engineers actually end up using the components

Agreed, which is mainly why I think we should start from a code-flexible but review-strict position.
